techlibs/microchip/synth_microchip.cc

```diff
@@ -0,0 +1,541 @@
+/*
+ISC License
+
+Copyright (C) 2024 Microchip Technology Inc. and its subsidiaries
+
+Permission to use, copy, modify, and/or distribute this software for any
+purpose with or without fee is hereby granted, provided that the above
+copyright notice and this permission notice appear in all copies.
+
+THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+
+#include "kernel/celltypes.h"
+#include "kernel/log.h"
+#include "kernel/register.h"
+#include "kernel/rtlil.h"
+
+USING_YOSYS_NAMESPACE
+PRIVATE_NAMESPACE_BEGIN
+
+struct SynthMicrochipPass : public ScriptPass {
+	SynthMicrochipPass() : ScriptPass("synth_microchip", "synthesis for Microchip FPGAs") {}
+
+	void help() override
+	{
+		//   |---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|
+		log("\n");
+		log("    synth_microchip [options]\n");
+		log("\n");
+		log("This command runs synthesis for Microchip FPGAs. Operating on\n");
+		log("partly selected designs is not supported (you must submit a fully-selected \n");
+		log("design). This command creates netlists that are compatible with Microchip \n");
+		log("PolarFire devices.\n");
+		log("\n");
+		log("    -top <module>\n");
+		log("        use the specified module as the top module\n");
+		log("\n");
+		log("    -family <family>\n");
+		log("        Run synthesis for the specified Microchip architecture. \n");
+		log("        Generate the synthesis netlist for the specified family.\n");
+		log("        supported values:\n");
+		log("        - polarfire: PolarFire\n");
+		log("\n");
+		log("    -edif <file>\n");
```

Comment:
I assume you're outputting to Libero for testing; what format does that normally use?

Reply:
There was very minimum testing through Libero. We only performed equivalence checks of post synthesis netlist (Verilog) against original netlist. 

I did try to push a very simple design through Libero. The format I used was Verilog for the input synthesized netlist.

But yeah, I guess it's probably a good idea to add -vlog option to dump Verilog.